src/components/common/Header.js

```diff
@@ -1,15 +1,11 @@
 import React, {PropTypes} from 'react';
 import { Link, IndexLink } from 'react-router';
 
-const Header = () => {
+const Header = (props, context) => {
```

Comment:
You dont need to get the `context` .

Also, you can use a shortcut to get the specific props:
`const Header = ({ toggleShowItem, showItem }) => {`